pkg/v1/providers/ytt/03_customizations/02_avi/ako-deployment.lib.yaml

```diff
@@ -0,0 +1,673 @@
+#@ load("@ytt:data", "data")
+#@ load("@ytt:base64", "base64")
+#@ load("@ytt:json", "json")
+#@ load("/lib/helpers.star", "get_default_tkr_bom_data", "get_default_tkg_bom_data", "tkg_image_repo")
+
+#@ akooRepo = get_default_tkr_bom_data().components["ako-operator"][0]
+#@ akoRepo = get_default_tkr_bom_data().components["load-balancer-and-ingress-service"][0]
+#@ rbacRepo = get_default_tkg_bom_data().components["kube_rbac_proxy"][0]
+
+#@ def ako_image_with_tag():
+#@  return "{}/{}:{}".format(tkg_image_repo(), akoRepo.images.loadBalancerAndIngressServiceImage.imagePath, akoRepo.images.loadBalancerAndIngressServiceImage.tag)
+#@ end
+
+#@ def tkg_system_namespace():
+#@  return "tkg-system"
+#@ end
+
+#@ def avi_system_namespace():
+#@  return "avi-system"
+#@ end
+
+#@ def cluster_namespaced_name():
+#@  return "{}-{}".format(tkg_system_namespace(), data.values.CLUSTER_NAME)
+#@ end
+
+#@ def ako_name():
+#@  return "ako-{}".format(cluster_namespaced_name())
+#@ end
+
+#! input cidr example 10.0.0.0/24
+#! output "10.0.0.0" and "24"
+#@ def network_cidr():
+#@  if data.values.AVI_MANAGEMENT_CLUSTER_VIP_NETWORK_NAME and data.values.AVI_MANAGEMENT_CLUSTER_VIP_NETWORK_CIDR:
+#@    cidr = data.values.AVI_MANAGEMENT_CLUSTER_VIP_NETWORK_CIDR
+#@  else:
+#@    cidr = data.values.AVI_DATA_NETWORK_CIDR
+#@  end
+#@  return cidr
+#@ end
+
+#@ def network_name():
+#@  if data.values.AVI_MANAGEMENT_CLUSTER_VIP_NETWORK_NAME and data.values.AVI_MANAGEMENT_CLUSTER_VIP_NETWORK_CIDR:
+#@    name = data.values.AVI_MANAGEMENT_CLUSTER_VIP_NETWORK_NAME
+#@  else:
+#@    name = data.values.AVI_DATA_NETWORK
+#@  end
+#@  return name
+#@ end
+
+#@ def vip_network():
+#@  return json.encode([{ "networkName": network_name(), "cidr": network_cidr() }])
+#@ end
+
+#@ def ako_namespace():
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: #@ avi_system_namespace()
+#@ end
+
+#@ def avi_secret():
+apiVersion: v1
+kind: Secret
+metadata:
+  name: avi-secret
+  namespace: tkg-system
+type: Opaque
+data:
+  username: #@ base64.encode(data.values.AVI_USERNAME)
+  password: #@ base64.encode(data.values.AVI_PASSWORD)
+  certificateAuthorityData: #@ data.values.AVI_CA_DATA_B64
+#@ end
+
+
+#@ def ako_serviceaccount():
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: ako-sa
+  namespace: tkg-system
+#@ end
+
+#@ def ako_configmap():
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: avi-k8s-config
+  namespace: tkg-system
+data:
+  controllerIP: #@ data.values.AVI_CONTROLLER
+  controllerVersion: "18.2.10"
+  cniPlugin: ""
+  shardVSSize: #@ data.values.AVI_INGRESS_SHARD_VS_SIZE
```

Comment:
The goal here is to have a really "lite" version of AKO that can provide basic L4 support to deploy in the bootstrap cluster giving the bootstrap cluster has such limited resources. So we don't need to configure so many advanced fields here.

AFAIK, we don't need to set the following fields here:
```
shardVSSize
passthroughShardSize
servicesAPI
enableEVH
layer7Only
defaultDomain
noPGForSNI
enableRHI
advancedL4
autoFQDN
nsSyncLabelKey
nsSyncLabelValue
bgpPeerLabels
nodeNetworkList
```

Ande we don't even support the following fields for now:
```
tenantsPerCluster
tenantName
nsxtT1LR
```

And I also encourage you to explore that on the CRDs, do we really need to have `aviinfrasetting` and other CRDS here?

Reply:
All CRDs have been removed to keep it minimal.